documentation/rfc7700.md

```diff
@@ -0,0 +1,151 @@
+---
+title: rfc7700 IRC Casemapping
+layout: spec
+copyrights:
+  -
+    name: "Daniel Oaks"
+    period: "2016"
+    email: "daniel@danieloaks.net"
+---
+This document describes a unicode-aware casemapping for IRC, based on the recommendations in [RFC 7700](https://tools.ietf.org/html/rfc7700).
+
+Client and channel names in languages other than English is a much-desired feature. This casemapping allows for an extended set of characters while minimising the risks around allowing these characters, and avoiding the technical limitations of prior solutions to this problem.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](http://tools.ietf.org/html/rfc2119).
+
+
+## Casemapping
+
+The server indicates that this casemapping is in use by advertising the `"CASEMAPPING"` `RPL_ISUPPORT` parameter with the value `"rfc7700"`. An example of this is shown below:
+
+    :irc.example.com 005 dan CASEMAPPING=rfc7700 :are supported by this server"
+
+When the `rfc7700` casemapping is in use, names (and other messages from the IRCd where possible) MUST be sent using the [UTF-8](https://tools.ietf.org/html/rfc3629) encoding.
+
+The `rfc7700` casemapping uses the PRECIS [Nickname profile](https://tools.ietf.org/html/rfc7700#section-2) as defined in [Section 2 of RFC 7700](https://tools.ietf.org/html/rfc7700#section-2).
+
+
+### Order of Operations
+
+Names being prepared MUST apply the following rules in the order shown:
+
+1. Preperation using the PRECIS [Nickname profile](https://tools.ietf.org/html/rfc7700#section-2).
+2. Check for restricted characters.
+
+These steps MUST happen in the order shown, or else the restricted characters check may miss characters that should be legitimately restricted.
+
+If a name does contain a restricted character (whether disallowed by the [Nickname profile](https://tools.ietf.org/html/rfc7700#section-2.2) or this document), it MUST be rejected by the server and MUST NOT be propogated to other clients. This is done through the appropriate numeric for the command which tried to set or use the invalid name such as `ERR_ERRONEUSNICKNAME`, `ERR_NOSUCHCHANNEL`, or whichever numeric is most appropriate.
```

Comment:
Cool, I changed the other links above to use the named link syntax since they were all referring to the same URL. The link here and the link to rfc7700 up the top differ from the others since they're linking to different sections (and the url's only replicated once throughout the doc).